tests/integration/test_cli_derivatives.py

```diff
@@ -205,3 +206,98 @@ def test_custom_imaging_sessions_created_for_missing_session_labels(
 
     # Note: order of items does not matter for dict comparison
     assert custom_ses_completed_pipes == completed_pipes_for_missing_ses_sub
+
+
+def test_unrecognized_pipelines_and_versions_excluded_from_output(
+    runner,
+    test_data,
+    test_data_upload_path,
+    default_derivatives_output_path,
+    load_test_json,
+):
+    """
+    Test that when a subset of pipelines or versions from a processing status file are unrecognized,
+    they are excluded from the output JSONLD with informative warnings, without causing the derivatives command to fail.
+    """
+    with pytest.warns(UserWarning) as w:
+        result = runner.invoke(
+            bagel,
+            [
+                "derivatives",
+                "-t",
+                test_data / "proc_status_unrecognized_pipelines.tsv",
+                "-p",
+                test_data_upload_path / "example_synthetic.jsonld",
+                "-o",
+                default_derivatives_output_path,
+            ],
+            catch_exceptions=False,
+        )
+
+    assert result.exit_code == 0, f"Errored out. STDOUT: {result.output}"
+
+    assert len(w) == 2
+    warnings = [warning.message.args[0] for warning in w]
+    for warn_substrings in [
+        ("unrecognized pipelines", "unknown-pipeline"),
+        ("unrecognized versions", "{'fmriprep': ['unknown.version']}"),
+    ]:
+        assert any(
+            all(substr in warning for substr in warn_substrings)
+            for warning in warnings
+        )
```

Comment:
I agree here, this section is honestly very hard to read. The other for loops below look fine - we may need to find better ways to unravel our jsonld files for testing... But here I am spending substantial time trying to understand what is being tested. 

How about just

```suggestion
    for warning in warnings:
        assert "unrecognized pipelines" in warning
        assert "unknown-pipeline" in warning
        assert "unrecognized versions" in warning
        assert "{'fmriprep': ['unknown.version']}" in warning
```

If I get this correctly, that's what you are asserting?

Reply:
ah, good bot. I indeed missed that we're testing that every warning contains **either** one of two sets of strings. 

I guess one question is: why do we need to test the exact wording of both substrings here since that's presumably already done in the unit tests. But one way of making this more readable is maybe

```suggestion
    for warning in warnings:
        assert (
        "unrecognized pipelines" in warning and "unknown-pipeline" in warning
        or 
        "unrecognized versions" in warning and "{'fmriprep': ['unknown.version']}" in warning
        )
```